default build options fail miserably

Why does `./configure` (or equivalently `./configure --prefix=/opt/shmem CC=gcc-8 FC=gfortran-8 CXX=g++-8` in my case)...
* fail to detect the `memcpy` transport?
* lead to a non-functioning library?

It is a bad thing if `./configure && make` creates a useless library.  Something should file before the tests run.  Either configure or make should fail on "no transport detected".

# highlights

```
Transports:
  Portals 4:      no
  OFI:            no
  XPMEM:          no
  CMA:            no
  memcpy:         no
```

```
# TOTAL: 115
# PASS:  0
# SKIP:  0
# XFAIL: 0
# FAIL:  115
# XPASS: 0
# ERROR: 0
```

# configure
```
jrhammon-mac01:sandia-shmem jrhammon$ ./configure --prefix=/opt/shmem CC=gcc-8 FC=gfortran-8 CXX=g++-8
checking for a BSD-compatible install... /usr/local/bin/ginstall -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /usr/local/bin/gmkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking whether UID '502' is supported by ustar format... yes
checking whether GID '20' is supported by ustar format... yes
checking how to create a ustar tar archive... gnutar
checking whether make supports nested variables... (cached) yes
checking build system type... x86_64-apple-darwin16.7.0
checking host system type... x86_64-apple-darwin16.7.0
checking whether make supports the include directive... yes (GNU style)
checking for gcc... gcc-8
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc-8 accepts -g... yes
checking for gcc-8 option to accept ISO C89... none needed
checking whether gcc-8 understands -c and -o together... yes
checking dependency style of gcc-8... gcc3
checking for grep that handles long lines and -e... /usr/bin/grep
checking for __attribute__... yes
checking for __attribute__(aligned)... yes
checking for __attribute__(always_inline)... yes
checking for __attribute__(cold)... yes
checking for __attribute__(const)... yes
checking for __attribute__(deprecated)... yes
checking for __attribute__(deprecated_argument)... yes
checking for __attribute__(format)... no
checking for __attribute__(format_funcptr)... no
checking for __attribute__(hot)... yes
checking for __attribute__(malloc)... yes
checking for __attribute__(may_alias)... yes
checking for __attribute__(no_instrument_function)... yes
checking for __attribute__(noinline)... yes
checking for __attribute__(nonnull)... no
checking for __attribute__(noreturn)... yes
checking for __attribute__(noreturn_funcptr)... yes
checking for __attribute__(packed)... yes
checking for __attribute__(pure)... yes
checking for __attribute__(sentinel)... no
checking for __attribute__(unused)... yes
checking for __attribute__(visibility)... yes
checking for __attribute__(warn_unused_result)... yes
checking for __attribute__(weak_alias)... no
checking for __attribute__(destructor)... yes
checking how to run the C preprocessor... gcc-8 -E
checking if user requested external PMIx support()... no
checking For total data ordering behavior... assume never available
checking for egrep... /usr/bin/grep -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking if C compiler and POSIX threads work as is... yes
checking for pthread_mutexattr_setpshared... yes
checking for pthread_condattr_setpshared... yes
checking for PTHREAD_MUTEX_ERRORCHECK_NP... no
checking for PTHREAD_MUTEX_ERRORCHECK... yes
checking for working POSIX threads package... yes
checking stdatomic.h usability... yes
checking stdatomic.h presence... yes
checking for stdatomic.h... yes
checking SYS_gettid syscall definitions... yes
checking for gcc-8 option to support OpenMP... -fopenmp
checking for gcc... (cached) gcc-8
checking whether we are using the GNU C compiler... (cached) yes
checking whether gcc-8 accepts -g... (cached) yes
checking for gcc-8 option to accept ISO C89... (cached) none needed
checking whether gcc-8 understands -c and -o together... (cached) yes
checking dependency style of gcc-8... (cached) gcc3
checking for inline... inline
checking whether we are using the GNU C++ compiler... yes
checking whether g++-8 accepts -g... yes
checking dependency style of g++-8... gcc3
checking for ar... ar
checking the archiver (ar) interface... ar
checking for perl... /usr/local/bin/perl
checking whether g++-8 supports C++11 features by default... yes
checking if -std=gnu11 works... yes
checking for m4... /usr/bin/m4
checking whether we are using the GNU Fortran compiler... yes
checking whether gfortran-8 accepts -g... yes
checking how to get verbose linking output from gfortran-8... -v
checking for Fortran libraries of gfortran-8...  -L/opt/intel/compilers_and_libraries_2018.3.185/mac/ipp/lib -L/opt/intel/compilers_and_libraries_2018.3.185/mac/compiler/lib -L/opt/intel/compilers_and_libraries_2018.3.185/mac/mkl/lib -L/opt/intel/compilers_and_libraries_2018.3.185/mac/tbb/lib -L/opt/intel/compilers_and_libraries_2018.3.185/mac/daal/lib -L/opt/intel/compilers_and_libraries_2018.3.185/mac/daal/../tbb/lib -L/usr/local/Cellar/gcc/8.2.0/lib/gcc/8/gcc/x86_64-apple-darwin16.7.0/8.2.0 -L/usr/local/Cellar/gcc/8.2.0/lib/gcc/8/gcc/x86_64-apple-darwin16.7.0/8.2.0/../../.. -lgfortran -lquadmath -lm
checking for dummy main to link with Fortran libraries... none
checking for Fortran name-mangling scheme... lower case, underscore, no extra underscore
checking for __builtin_trap... yes
checking for library containing clock_gettime... none required
checking sched_getaffinity definition... no
checking --with-portals4 value... simple ok (unspecified)
checking --with-portals4-libdir value... simple ok (unspecified)
checking portals4.h usability... no
checking portals4.h presence... no
checking for portals4.h... no
checking --with-ofi value... simple ok (unspecified)
checking --with-ofi-libdir value... simple ok (unspecified)
checking looking for OFI libfabric in... ()
checking rdma/fabric.h usability... yes
checking rdma/fabric.h presence... yes
checking for rdma/fabric.h... yes
looking for library without search path
checking for library containing fi_getinfo... -lfabric
configure: WARNING: Neither OFI nor portals4 transport requested
Neither XPMEM nor CMA transport requested
checking if user requested PMI support... no
checking fnmatch.h usability... yes
checking fnmatch.h presence... yes
checking for fnmatch.h... yes
checking size of char... 1
checking size of short... 2
checking size of int... 4
checking size of long... 8
checking size of long long... 8
checking size of ptrdiff_t... 8
checking size of unsigned int... 4
checking size of unsigned long... 8
checking size of unsigned long long... 8
checking size of size_t... 8
checking size of float... 4
checking size of double... 8
checking size of long double... 16
checking size of void*... 8
checking size of SHMEM_BCAST_SYNC_SIZE... 1
checking size of SHMEM_REDUCE_SYNC_SIZE... 35
checking size of SHMEM_BARRIER_SYNC_SIZE... 16
checking size of SHMEM_COLLECT_SYNC_SIZE... 18
checking size of SHMEM_ALLTOALL_SYNC_SIZE... 16
checking size of SHMEM_ALLTOALLS_SYNC_SIZE... 16
checking size of SHMEM_SYNC_SIZE... 35
checking size of SHMEM_REDUCE_MIN_WRKDATA_SIZE... 1
checking if long can hold lock_t... yes
checking if complex sizes are appropriate... yes
checking if Fortran compiler supports ISO_C_BINDING... yes
checking size of Fortran LOGICAL... 4
checking size of Fortran CHARACTER... 1
checking size of Fortran INTEGER... 4
checking size of Fortran REAL... 4
checking size of Fortran DOUBLE PRECISION... 8
checking size of Fortran COMPLEX... 8
checking size of Fortran DOUBLE COMPLEX... 16
checking for C type corresponding to INTEGER... int
checking for C type corresponding to DOUBLE PRECISION... double
checking KIND value of Fortran C_INT32_T... 4
checking KIND value of Fortran C_INT64_T... 8
checking KIND value of Fortran C_FLOAT... 4
checking KIND value of Fortran C_DOUBLE... 8
checking KIND value of Fortran C_FLOAT_COMPLEX... 4
checking KIND value of Fortran C_DOUBLE_COMPLEX... 8
checking if gcc-8 supports -fvisibility=hidden... yes
checking whether to enable symbol visibility... yes (via -fvisibility=hidden)
checking for data segment pointer... found
checking how to print strings... printf
checking for a sed that does not truncate output... /usr/local/bin/gsed
checking for fgrep... /usr/bin/grep -F
checking for ld used by gcc-8... /usr/bin/ld
checking if the linker (/usr/bin/ld) is GNU ld... no
checking for BSD- or MS-compatible name lister (nm)... /usr/bin/nm -B
checking the name lister (/usr/bin/nm -B) interface... BSD nm
checking whether ln -s works... yes
checking the maximum length of command line arguments... 196608
checking how to convert x86_64-apple-darwin16.7.0 file names to x86_64-apple-darwin16.7.0 format... func_convert_file_noop
checking how to convert x86_64-apple-darwin16.7.0 file names to toolchain format... func_convert_file_noop
checking for /usr/bin/ld option to reload object files... -r
checking for objdump... objdump
checking how to recognize dependent libraries... pass_all
checking for dlltool... no
checking how to associate runtime and link libraries... printf %s\n
checking for archiver @FILE support... no
checking for strip... strip
checking for ranlib... ranlib
checking command to parse /usr/bin/nm -B output from gcc-8 object... ok
checking for sysroot... no
checking for a working dd... /bin/dd
checking how to truncate binary pipes... /bin/dd bs=4096 count=1
checking for mt... no
checking if : is a manifest tool... no
checking for dsymutil... dsymutil
checking for nmedit... nmedit
checking for lipo... lipo
checking for otool... otool
checking for otool64... no
checking for -single_module linker flag... yes
checking for -exported_symbols_list linker flag... yes
checking for -force_load linker flag... yes
checking for dlfcn.h... yes
checking for objdir... .libs
checking if gcc-8 supports -fno-rtti -fno-exceptions... no
checking for gcc-8 option to produce PIC... -fno-common -DPIC
checking if gcc-8 PIC flag -fno-common -DPIC works... yes
checking if gcc-8 static flag -static works... no
checking if gcc-8 supports -c -o file.o... yes
checking if gcc-8 supports -c -o file.o... (cached) yes
checking whether the gcc-8 linker (/usr/bin/ld) supports shared libraries... yes
checking dynamic linker characteristics... darwin16.7.0 dyld
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... yes
checking how to run the C++ preprocessor... g++-8 -E
checking for ld used by g++-8... /usr/bin/ld
checking if the linker (/usr/bin/ld) is GNU ld... no
checking whether the g++-8 linker (/usr/bin/ld) supports shared libraries... yes
checking for g++-8 option to produce PIC... -fno-common -DPIC
checking if g++-8 PIC flag -fno-common -DPIC works... yes
checking if g++-8 static flag -static works... no
checking if g++-8 supports -c -o file.o... yes
checking if g++-8 supports -c -o file.o... (cached) yes
checking whether the g++-8 linker (/usr/bin/ld) supports shared libraries... yes
checking dynamic linker characteristics... darwin16.7.0 dyld
checking how to hardcode library paths into programs... immediate
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... yes
checking for gfortran-8 option to produce PIC... -fno-common
checking if gfortran-8 PIC flag -fno-common works... yes
checking if gfortran-8 static flag -static works... no
checking if gfortran-8 supports -c -o file.o... yes
checking if gfortran-8 supports -c -o file.o... (cached) yes
checking whether the gfortran-8 linker (/usr/bin/ld) supports shared libraries... yes
checking dynamic linker characteristics... darwin16.7.0 dyld
checking how to hardcode library paths into programs... immediate
checking if build filesystem is case sensitive... no
configure: creating ./config.lt
config.lt: creating libtool
checking if linker supports RPATH... yes (no extra flags needed)
checking for OPAL LDFLAGS...   -Wl,-no_pie
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating Makefile
config.status: creating sandia-openshmem.spec
config.status: creating man/Makefile
config.status: creating bindings/Makefile
config.status: creating pmi-simple/Makefile
config.status: creating mpp/Makefile
config.status: creating mpp/shmem.fh
config.status: creating mpp/shmemx.fh
config.status: creating mpp/shmem.h4
config.status: creating mpp/shmem_c_func.h4
config.status: creating mpp/shmemx_c_func.h4
config.status: creating mpp/shmemx.h4
config.status: creating src/Makefile
config.status: creating src/sandia-openshmem.pc
config.status: creating test/Makefile
config.status: creating test/unit/Makefile
config.status: creating test/shmemx/Makefile
config.status: creating test/include/Makefile
config.status: creating test/performance/Makefile
config.status: creating test/performance/shmem_perf_suite/Makefile
config.status: creating test/performance/tests/Makefile
config.status: creating test/apps/Makefile
config.status: creating src/config.h
config.status: executing depfiles commands
config.status: executing libtool commands
configure: WARNING: No PMI found, if build fails consider --enable-pmi-simple or --with-pmi
configure: WARNING: No transport found
configure: WARNING: Fortran bindings have been deprecated as of OpenSHMEM v1.4

Compilers:
  C:              gcc-8
  C++:            g++-8
  Fortran:        gfortran-8

Transports:
  Portals 4:      no
  OFI:            no
  XPMEM:          no
  CMA:            no
  memcpy:         no

Global Options:
  Remote VA:      no
  Thread support: yes
  Error checking: no

```

# make
```
jrhammon-mac01:sandia-shmem jrhammon$ make -j4
Making all in bindings
  CC       generate_bind_c11_extras.o
  CXX      generate_bind_cxx_extras.o
  CCLD     generate_bind_c11_extras
  CXXLD    generate_bind_cxx_extras
  GEN      shmem_bind_c11_extras.m4
  GEN      shmem_bind_cxx_extras.m4
/Applications/Xcode.app/Contents/Developer/usr/bin/make  all-am
make[2]: Nothing to be done for `all-am'.
Making all in man
make[1]: Nothing to be done for `all'.
Making all in mpp
  GEN      shmem.h
  GEN      shmemx.h
/Applications/Xcode.app/Contents/Developer/usr/bin/make  all-am
make[2]: Nothing to be done for `all-am'.
Making all in pmi-simple
make[1]: Nothing to be done for `all'.
Making all in src
  GEN      atomic_c.c
  GEN      collectives_c.c
  GEN      data_c.c
  GEN      synchronization_c.c
  GEN      collectives_f.c
  GEN      data_f.c
  GEN      build_info.h
/Applications/Xcode.app/Contents/Developer/usr/bin/make  all-am
  CC       malloc.lo
  CC       shmem_free_list.lo
  CC       init.lo
  CC       collectives.lo
  CC       init_c.lo
  CC       query_c.lo
  CC       accessibility_c.lo
  CC       symmetric_heap_c.lo
  CC       remote_pointer_c.lo
  CC       lock_c.lo
  CC       cache_management_c.lo
  CC       util.lo
  CC       shmem_env.lo
  CC       contexts_c.lo
  CC       contexts.lo
  CC       perf_counters_c.lo
  CC       init_f.lo
  CC       query_f.lo
  CC       accessibility_f.lo
  CC       symmetric_heap_f.lo
  CC       remote_pointer_f.lo
  CC       atomic_f.lo
  CC       synchronization_f.lo
  CC       lock_f.lo
  CC       cache_management_f.lo
  CC       transport_none.lo
  CC       atomic_c.lo
  CC       collectives_c.lo
  CC       data_c.lo
  CC       synchronization_c.lo
  CC       collectives_f.lo
  CC       data_f.lo
  SED      oshcc
  SED      oshrun
  SED      oshc++
  SED      oshfort
  CCLD     libsma.la
Making all in test
Making all in unit
make[2]: Nothing to be done for `all'.
Making all in shmemx
make[2]: Nothing to be done for `all'.
Making all in performance
Making all in shmem_perf_suite
make[3]: Nothing to be done for `all'.
Making all in tests
make[3]: Nothing to be done for `all'.
make[3]: Nothing to be done for `all-am'.
Making all in apps
make[2]: Nothing to be done for `all'.
Making all in include
make[2]: Nothing to be done for `all'.
make[2]: Nothing to be done for `all-am'.
make[1]: Nothing to be done for `all-am'.
```

# make check
```
jrhammon-mac01:sandia-shmem jrhammon$ make check
Making check in bindings
/Applications/Xcode.app/Contents/Developer/usr/bin/make  check-am
make[2]: Nothing to be done for `check-am'.
Making check in man
make[1]: Nothing to be done for `check'.
Making check in mpp
/Applications/Xcode.app/Contents/Developer/usr/bin/make  check-am
make[2]: Nothing to be done for `check-am'.
Making check in pmi-simple
make[1]: Nothing to be done for `check'.
Making check in src
/Applications/Xcode.app/Contents/Developer/usr/bin/make  check-am
make[2]: Nothing to be done for `check-am'.
Making check in test
Making check in unit
/Applications/Xcode.app/Contents/Developer/usr/bin/make  hello pi micro_unit_shmem circular_shift accessible_ping max_reduction big_reduction to_all strided_put barrier bcast put1 get1 swap1 ping pingpong shmalloc shmem_calloc shrealloc shmemalign get_g iput32 iput64 iput128 iput_short iput_double iput_float iput_long iput_longdouble iput_longlong sping pingpong-short ipgm iput-iget swapm cswap waituntil atomic_inc set_lock test_lock test_lock_cswap fcollect64 bigput bigget ns bcast_flood lfinc shmem_info query_thread asym_alloc set_fetch alltoall alltoalls c11_test_shmem_g c11_test_shmem_get c11_test_shmem_p c11_test_shmem_put c11_test_shmem_atomic_fetch c11_test_shmem_atomic_set c11_test_shmem_atomic_add c11_test_shmem_atomic_inc c11_test_shmem_atomic_and c11_test_shmem_atomic_or c11_test_shmem_atomic_xor c11_test_shmem_atomic_swap c11_test_shmem_atomic_cswap c11_test_shmem_wait_until c11_test_shmem_test get_nbi put_nbi rma_coverage collect repeated_barriers repeated_syncs broadcast_active_set reduce_active_set collect_active_set atomic_bitwise nop_collectives self_collectives zero_comm sync-size shmem_ctx_pipelined_reduce many-ctx shmem_test global_exit  cxx_test_shmem_complex cxx_test_shmem_g cxx_test_shmem_get cxx_test_shmem_p cxx_test_shmem_put cxx_test_shmem_atomic_fetch cxx_test_shmem_atomic_set cxx_test_shmem_atomic_add cxx_test_shmem_atomic_inc cxx_test_shmem_atomic_and cxx_test_shmem_atomic_or cxx_test_shmem_atomic_xor cxx_test_shmem_atomic_swap cxx_test_shmem_atomic_cswap cxx_test_shmem_wait_until cxx_test_shmem_test hello_f complex_reductions_f set_fetch_f shmem_info_f mt_a2a mt_contention mt_membar mt_lock_test threading web thread_wait query_thread_funneled shmem_ctx
  CC       hello.o
  CCLD     hello
  CC       pi.o
  CCLD     pi
  CC       micro_unit_shmem.o
  CCLD     micro_unit_shmem
  CC       circular_shift.o
  CCLD     circular_shift
  CC       accessible_ping.o
  CCLD     accessible_ping
  CC       max_reduction.o
  CCLD     max_reduction
  CC       big_reduction.o
  CCLD     big_reduction
  CC       to_all.o
  CCLD     to_all
  CC       strided_put.o
  CCLD     strided_put
  CC       barrier.o
  CCLD     barrier
  CC       bcast.o
  CCLD     bcast
  CC       put1.o
  CCLD     put1
  CC       get1.o
  CCLD     get1
  CC       swap1.o
  CCLD     swap1
  CC       ping.o
  CCLD     ping
  CC       pingpong.o
  CCLD     pingpong
  CC       shmalloc.o
  CCLD     shmalloc
  CC       shmem_calloc.o
  CCLD     shmem_calloc
  CC       shrealloc.o
  CCLD     shrealloc
  CC       shmemalign.o
  CCLD     shmemalign
  CC       get_g.o
  CCLD     get_g
  CC       iput32.o
  CCLD     iput32
  CC       iput64.o
  CCLD     iput64
  CC       iput128.o
  CCLD     iput128
  CC       iput_short.o
  CCLD     iput_short
  CC       iput_double.o
  CCLD     iput_double
  CC       iput_float.o
  CCLD     iput_float
  CC       iput_long.o
  CCLD     iput_long
  CC       iput_longdouble.o
  CCLD     iput_longdouble
  CC       iput_longlong.o
  CCLD     iput_longlong
  CC       sping.o
  CCLD     sping
  CC       pingpong-short.o
  CCLD     pingpong-short
  CC       ipgm.o
  CCLD     ipgm
  CC       iput-iget.o
  CCLD     iput-iget
  CC       swapm.o
  CCLD     swapm
  CC       cswap.o
  CCLD     cswap
  CC       waituntil.o
  CCLD     waituntil
  CC       atomic_inc.o
  CCLD     atomic_inc
  CC       set_lock.o
  CCLD     set_lock
  CC       test_lock.o
  CCLD     test_lock
  CC       test_lock_cswap.o
  CCLD     test_lock_cswap
  CC       fcollect64.o
  CCLD     fcollect64
  CC       bigput.o
  CCLD     bigput
  CC       bigget.o
  CCLD     bigget
  CC       ns.o
  CCLD     ns
  CC       bcast_flood.o
  CCLD     bcast_flood
  CC       lfinc.o
  CCLD     lfinc
  CC       shmem_info.o
  CCLD     shmem_info
  CC       query_thread.o
  CCLD     query_thread
  CC       asym_alloc.o
  CCLD     asym_alloc
  CC       set_fetch.o
  CCLD     set_fetch
  CC       alltoall.o
  CCLD     alltoall
  CC       alltoalls.o
  CCLD     alltoalls
  CC       c11_test_shmem_g.o
  CCLD     c11_test_shmem_g
  CC       c11_test_shmem_get.o
  CCLD     c11_test_shmem_get
  CC       c11_test_shmem_p.o
  CCLD     c11_test_shmem_p
  CC       c11_test_shmem_put.o
  CCLD     c11_test_shmem_put
  CC       c11_test_shmem_atomic_fetch.o
  CCLD     c11_test_shmem_atomic_fetch
  CC       c11_test_shmem_atomic_set.o
  CCLD     c11_test_shmem_atomic_set
  CC       c11_test_shmem_atomic_add.o
  CCLD     c11_test_shmem_atomic_add
  CC       c11_test_shmem_atomic_inc.o
  CCLD     c11_test_shmem_atomic_inc
  CC       c11_test_shmem_atomic_and.o
  CCLD     c11_test_shmem_atomic_and
  CC       c11_test_shmem_atomic_or.o
  CCLD     c11_test_shmem_atomic_or
  CC       c11_test_shmem_atomic_xor.o
  CCLD     c11_test_shmem_atomic_xor
  CC       c11_test_shmem_atomic_swap.o
  CCLD     c11_test_shmem_atomic_swap
  CC       c11_test_shmem_atomic_cswap.o
  CCLD     c11_test_shmem_atomic_cswap
  CC       c11_test_shmem_wait_until.o
  CCLD     c11_test_shmem_wait_until
  CC       c11_test_shmem_test.o
  CCLD     c11_test_shmem_test
  CC       get_nbi.o
  CCLD     get_nbi
  CC       put_nbi.o
  CCLD     put_nbi
  CC       rma_coverage.o
  CCLD     rma_coverage
  CC       collect.o
  CCLD     collect
  CC       repeated_barriers.o
  CCLD     repeated_barriers
  CC       repeated_syncs.o
  CCLD     repeated_syncs
  CC       broadcast_active_set.o
  CCLD     broadcast_active_set
  CC       reduce_active_set.o
  CCLD     reduce_active_set
  CC       collect_active_set.o
  CCLD     collect_active_set
  CC       atomic_bitwise.o
  CCLD     atomic_bitwise
  CC       nop_collectives.o
  CCLD     nop_collectives
  CC       self_collectives.o
  CCLD     self_collectives
  CC       zero_comm.o
  CCLD     zero_comm
  CC       sync-size.o
  CCLD     sync-size
  CC       shmem_ctx_pipelined_reduce.o
  CCLD     shmem_ctx_pipelined_reduce
  CC       many-ctx.o
  CCLD     many-ctx
  CC       shmem_test.o
  CCLD     shmem_test
  CC       global_exit.o
  CCLD     global_exit
  CXX      cxx_test_shmem_complex.o
  CXXLD    cxx_test_shmem_complex
  CXX      cxx_test_shmem_g.o
  CXXLD    cxx_test_shmem_g
  CXX      cxx_test_shmem_get.o
  CXXLD    cxx_test_shmem_get
  CXX      cxx_test_shmem_p.o
  CXXLD    cxx_test_shmem_p
  CXX      cxx_test_shmem_put.o
  CXXLD    cxx_test_shmem_put
  CXX      cxx_test_shmem_atomic_fetch.o
  CXXLD    cxx_test_shmem_atomic_fetch
  CXX      cxx_test_shmem_atomic_set.o
  CXXLD    cxx_test_shmem_atomic_set
  CXX      cxx_test_shmem_atomic_add.o
  CXXLD    cxx_test_shmem_atomic_add
  CXX      cxx_test_shmem_atomic_inc.o
  CXXLD    cxx_test_shmem_atomic_inc
  CXX      cxx_test_shmem_atomic_and.o
  CXXLD    cxx_test_shmem_atomic_and
  CXX      cxx_test_shmem_atomic_or.o
  CXXLD    cxx_test_shmem_atomic_or
  CXX      cxx_test_shmem_atomic_xor.o
  CXXLD    cxx_test_shmem_atomic_xor
  CXX      cxx_test_shmem_atomic_swap.o
  CXXLD    cxx_test_shmem_atomic_swap
  CXX      cxx_test_shmem_atomic_cswap.o
  CXXLD    cxx_test_shmem_atomic_cswap
  CXX      cxx_test_shmem_wait_until.o
  CXXLD    cxx_test_shmem_wait_until
  CXX      cxx_test_shmem_test.o
  CXXLD    cxx_test_shmem_test
  FC       hello_f.o
  FCLD     hello_f
  FC       complex_reductions_f.o
  FCLD     complex_reductions_f
  FC       set_fetch_f.o
  FCLD     set_fetch_f
  FC       shmem_info_f.o
  FCLD     shmem_info_f
  CC       mt_a2a-mt_a2a.o
  CCLD     mt_a2a
  CC       mt_contention-mt_contention.o
  CCLD     mt_contention
  CC       mt_membar-mt_membar.o
  CCLD     mt_membar
  CC       mt_lock_test-mt_lock_test.o
  CC       mt_lock_test-mt_lock.o
  CCLD     mt_lock_test
  CC       threading-threading.o
  CCLD     threading
  CC       web-web.o
  CCLD     web
  CC       thread_wait-thread_wait.o
  CCLD     thread_wait
  CC       query_thread_funneled-query_thread.o
  CCLD     query_thread_funneled
  CC       shmem_ctx-shmem_ctx.o
  CCLD     shmem_ctx
/Applications/Xcode.app/Contents/Developer/usr/bin/make  check-TESTS
FAIL: hello
FAIL: pi
FAIL: micro_unit_shmem
FAIL: circular_shift
FAIL: accessible_ping
FAIL: max_reduction
FAIL: big_reduction
FAIL: to_all
FAIL: strided_put
FAIL: barrier
FAIL: bcast
FAIL: put1
FAIL: get1
FAIL: swap1
FAIL: ping
FAIL: pingpong
FAIL: shmalloc
FAIL: shmem_calloc
FAIL: shrealloc
FAIL: shmemalign
FAIL: get_g
FAIL: iput32
FAIL: iput64
FAIL: iput128
FAIL: iput_short
FAIL: iput_double
FAIL: iput_float
FAIL: iput_long
FAIL: iput_longdouble
FAIL: iput_longlong
FAIL: sping
FAIL: pingpong-short
FAIL: ipgm
FAIL: iput-iget
FAIL: swapm
FAIL: cswap
FAIL: waituntil
FAIL: atomic_inc
FAIL: set_lock
FAIL: test_lock
FAIL: test_lock_cswap
FAIL: fcollect64
FAIL: bigput
FAIL: bigget
FAIL: ns
FAIL: bcast_flood
FAIL: lfinc
FAIL: shmem_info
FAIL: query_thread
FAIL: asym_alloc
FAIL: set_fetch
FAIL: alltoall
FAIL: alltoalls
FAIL: c11_test_shmem_g
FAIL: c11_test_shmem_get
FAIL: c11_test_shmem_p
FAIL: c11_test_shmem_put
FAIL: c11_test_shmem_atomic_fetch
FAIL: c11_test_shmem_atomic_set
FAIL: c11_test_shmem_atomic_add
FAIL: c11_test_shmem_atomic_inc
FAIL: c11_test_shmem_atomic_and
FAIL: c11_test_shmem_atomic_or
FAIL: c11_test_shmem_atomic_xor
FAIL: c11_test_shmem_atomic_swap
FAIL: c11_test_shmem_atomic_cswap
FAIL: c11_test_shmem_wait_until
FAIL: c11_test_shmem_test
FAIL: get_nbi
FAIL: put_nbi
FAIL: rma_coverage
FAIL: collect
FAIL: repeated_barriers
FAIL: repeated_syncs
FAIL: broadcast_active_set
FAIL: reduce_active_set
FAIL: collect_active_set
FAIL: atomic_bitwise
FAIL: nop_collectives
FAIL: self_collectives
FAIL: zero_comm
FAIL: sync-size
FAIL: shmem_ctx_pipelined_reduce
FAIL: many-ctx
FAIL: shmem_test
FAIL: global_exit
FAIL: cxx_test_shmem_complex
FAIL: cxx_test_shmem_g
FAIL: cxx_test_shmem_get
FAIL: cxx_test_shmem_p
FAIL: cxx_test_shmem_put
FAIL: cxx_test_shmem_atomic_fetch
FAIL: cxx_test_shmem_atomic_set
FAIL: cxx_test_shmem_atomic_add
FAIL: cxx_test_shmem_atomic_inc
FAIL: cxx_test_shmem_atomic_and
FAIL: cxx_test_shmem_atomic_or
FAIL: cxx_test_shmem_atomic_xor
FAIL: cxx_test_shmem_atomic_swap
FAIL: cxx_test_shmem_atomic_cswap
FAIL: cxx_test_shmem_wait_until
FAIL: cxx_test_shmem_test
FAIL: hello_f
FAIL: complex_reductions_f
FAIL: set_fetch_f
FAIL: shmem_info_f
FAIL: mt_a2a
FAIL: mt_contention
FAIL: mt_membar
FAIL: mt_lock_test
FAIL: threading
FAIL: web
FAIL: thread_wait
FAIL: query_thread_funneled
FAIL: shmem_ctx
============================================================================
Testsuite summary for Sandia OpenSHMEM 1.4.2
============================================================================
# TOTAL: 115
# PASS:  0
# SKIP:  0
# XFAIL: 0
# FAIL:  115
# XPASS: 0
# ERROR: 0
============================================================================
See test/unit/test-suite.log
Please report to https://github.com/Sandia-OpenSHMEM/SOS/issues
============================================================================
make[4]: *** [test-suite.log] Error 1
make[3]: *** [check-TESTS] Error 2
make[2]: *** [check-am] Error 2
make[1]: *** [check-recursive] Error 1
make: *** [check-recursive] Error 1
```

--- a/configure.ac
+++ b/configure.ac
@@ -782,10 +782,10 @@ fi
 AC_OUTPUT
 
 AS_IF([test "$enable_pmi_mpi" != "yes" -a "$enable_pmi_simple" != "yes" -a "$opal_external_pmix_version_found" != 1 -a "$ompi_check_pmi_happy" != "yes" -a -z "$pmi_type"],
-         [AC_MSG_WARN([No PMI found, if build fails consider --enable-pmi-simple or --with-pmi])])
+         [AC_MSG_ERROR([No PMI client interface was configured])])
 
-AS_IF([test "$transport_portals4" != "yes" -a "$transport_xpmem" != "yes" -a "$transport_cma" != "yes" -a "$transport_ofi" != "yes"],
-      [AC_MSG_WARN([No transport found])])
+AS_IF([test "$transport_portals4" != "yes" -a "$transport_ofi" != "yes"],
+      [AC_MSG_WARN([No transport found, resulting library be unable to exchange messages])])
 
 AS_IF([test "$shmem_cv_c11_works" != "yes"],
       [AC_MSG_WARN([C compiler does not support _Generic, unabled to verify and test C11 bindings])])
@@ -825,9 +825,11 @@ echo ""
 echo "Transports:"
 echo "  Portals 4:      $transport_portals4"
 echo "  OFI:            $transport_ofi"
+echo ""
+echo "On Node Communication:"
 echo "  XPMEM:          $transport_xpmem"
 echo "  CMA:            $transport_cma"
-echo "  memcpy:         $transport_memcpy"
+echo "  memcpy (self):  $transport_memcpy"
 echo ""
 echo "Global Options:"
 if test "$enable_remote_virtual_addressing" = "yes"; then

--- a/configure.ac
+++ b/configure.ac
@@ -782,10 +782,10 @@ fi
 AC_OUTPUT
 
 AS_IF([test "$enable_pmi_mpi" != "yes" -a "$enable_pmi_simple" != "yes" -a "$opal_external_pmix_version_found" != 1 -a "$ompi_check_pmi_happy" != "yes" -a -z "$pmi_type"],
-         [AC_MSG_ERROR([No PMI client interface was configured])])
+         [AC_MSG_ERROR([No PMI client interface was configured, consider --enable-pmi-simple or --with-pmi])])
 
 AS_IF([test "$transport_portals4" != "yes" -a "$transport_ofi" != "yes"],
-      [AC_MSG_WARN([No transport found, resulting library be unable to exchange messages])])
+      [AC_MSG_WARN([No transport found, resulting library will be unable to exchange messages])])
 
 AS_IF([test "$shmem_cv_c11_works" != "yes"],
       [AC_MSG_WARN([C compiler does not support _Generic, unabled to verify and test C11 bindings])])

--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -118,22 +118,22 @@ if USE_PMI_SIMPLE
 AM_CPPFLAGS += -I$(top_srcdir)/pmi-simple
 libsma_la_SOURCES += \
 	runtime-pmi.c
-endif
-
+else
 if USE_PMI_MPI
 libsma_la_SOURCES += \
 	runtime-mpi.c
-endif
-
+else
 if USE_PMI2
 libsma_la_SOURCES += \
 	runtime-pmi2.c
-endif
-
+else
 if USE_PMIX
 libsma_la_SOURCES += \
 	runtime-pmix.c
-endif
+endif USE_PMIX
+endif !USE_PMI2
+endif !USE_PMI_MPI
+endif !USE_PMI_SIMPLE
 
 pkgconfig_DATA = sandia-openshmem.pc
 

